R/compareScenarios.R

```diff
@@ -171,28 +171,28 @@ compareScenarios <- function(mif, hist,
   
   # print a warning message on repeated names
   uniqueScenarioName = unique(unlist(mifScenarios, use.names = FALSE))
+  
   if (length(uniqueScenarioName) < length(mifScenarios)){
     warning(paste0("There are repeated scenario names in comparison! Appending numerical value to repeated names"))
-  }
-  
-  # append numerical value to repeated names
-  mifScenarios_new <- c()
-  for(i in 1:length(uniqueScenarioName)){
-    unique_element = uniqueScenarioName[[i]]
-    k = 1
-    for(j in 1:length(mifScenarios)){
-      if (mifScenarios[[j]] == unique_element){
-        mifScenarios_new <-c(mifScenarios_new, paste0(unique_element, "_", k))
-        k = k + 1
-      }
+ 
+    #get duplicated names
+    dup_names = mifScenarios[duplicated(mifScenarios)]
+    
+    #append a numerical value to the repeated names
+    for (i in 1:length(dup_names)){
+      dup_name = dup_names[[i]]
+      #number of duplicates
+      dupnum <- seq(1,length(mifScenarios[mifScenarios == dup_name]))
+      mifScenarios[mifScenarios == dup_name] <- paste0(mifScenarios[mifScenarios == dup_name], "_", dupnum)  
+    }
+    
+    mifScenarios_new <- mifScenarios
+    #rename scenario names
+    for(i in 1:length(mifScenarios)){
+      magclass::getNames(mifData[[i]],dim=1) <- mifScenarios_new[[i]]
```

Comment:
why is `magclass::` used here? not needed

Reply:
I used this "magclass::getNames" originally since many lines below also contain "magclass::getNames" (but some other times only getNames is used). I am not sure which is correct/needed. Should I only remove this one or all the subsequent ones also (those lines are not from me)?

Reply:
feel free to remove only yours :)

Reply:
done :)